services/web/server/src/simcore_service_webserver/exporter/file_downloader.py

```diff
@@ -38,19 +27,16 @@ async def append_file(self, link: str, download_path: Path):
 
     async def download_files(self, app: Application):
         """starts the download and waits for all files to finish"""
-
-        # run this async, parfive will support aiofiles in the future as stated above
-        wrapped_function = aiofiles_os.wrap(self.downloader.download)
         exporter_settings = get_settings(app)
-        results = await wrapped_function(
+        results = await self.downloader.run_download(
             timeouts={
                 "total": exporter_settings.downloader_max_timeout_seconds,
                 "sock_read": 90,  # default as in parfive code
             }
         )
-        log.debug("Download results %s", results)
 
-        if len(results) != self.total_files_added:
-            raise ExporterException(
-                "Not all files were downloaded. Please check the logs above."
-            )
+        log.debug("Download %s using %s", f"{results=}", f"{self.downloader=}")
+        if len(results) != self.total_files_added or len(results.errors) > 0:
+            message = f"Not all files were downloaded see: {results.errors=}"
+            log.error(message)
```

Comment:
Since you are logging as ``error``, this means that this exception cannot be handled and the error mended e.g. with a retry?

IMO the log.error should happen wherever the exception is handled and not where it is raised. In special cases i would put here a log BUT in ``debug``

Reply:
same question here. what happens with this error? should it not be converted wherever the handler is?

Reply:
To be sure, I've put in a test. The errors from the downloader end up in the reply to the user as well.